### DISCUSSION TITLE
CodeQL setup and usefulness

### DISCUSSION THREAD

**Discussion by ihhub** [ihhub/fheroes2#9247] [General]
GitHub allows to setup CodeQL for repositories:
![image](https://github.com/user-attachments/assets/0ad192a6-53c8-44ae-a077-01635a35bdf5)

You can read about it [here](https://docs.github.com/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/about-code-scanning-with-codeql) and [here](https://docs.github.com/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/configuring-code-scanning-at-scale#eligible-repositories-for-codeql-default-setup).

According to the requirements and language support, we can enable CodeQL for our repository.

**Comment by oleg-derevenetz** [comment]
Hi @ihhub,

First of all, the "default" CodeQL setup will not work good enough with our setup (for instance, it will not able to install the pre-build deps for Android or build optional tools), we need the custom setup (I [tried](https://github.com/oleg-derevenetz/fheroes2/commit/958321a4c56ac97edc6d7f6ffedd0ed9370e0af7) it on this project BTW).

Second, I doubt that it will find something that clang-analyzer, clang-tidy and SonarQube would not have found already (I already used it in some other projects, and it did not do this). In addition, it approaches the case too formally, and this leads to false positives. For instance, it "found" the following "issue" in this project:

![issue](https://github.com/user-attachments/assets/2533c540-2adc-45b4-92b1-83da43508779)

It fails to see that this `File` instance is used only temporary to get the "name" part from it (the last name in the pathname's name sequence) and then this name is additionally checked to have a valid file extension, so it never can be `'..'`. I'd say it's a pretty clumsy job for a "security analyzer". It seems that it just has a few dozen simple rules, and it just flags all the places that fit these rules without trying to conduct any flow analysis.

**Comment by ihhub** [comment]
Thank you very much for the detailed analysis, @oleg-derevenetz !

We shall not then proceed with this tool.
